sqlmesh/core/snapshot/definition.py

```diff
@@ -614,14 +626,19 @@ def missing_intervals(
             start: The start date/time of the interval (inclusive)
             end: The end date/time of the interval (inclusive)
             latest: The date/time to use for latest (inclusive)
+            restatements: A set of snapshot names being restated
 
         Returns:
             A list of all the missing intervals as epoch timestamps.
         """
+        restatements = restatements or set()
         if self.is_symbolic or (self.is_seed and self.intervals):
             return []
 
         latest = make_inclusive_end(latest or now())
+        end = (
+            make_inclusive_end(now()) if self.depends_on_past and self.name in restatements else end
```

Comment:
should this be derived from `latest`? or otherwise should `latest` and `end` share the same value of `now()`?

Reply:
I think the fact that `end` can be greater than `latest` may lead to the kind of edge cases I don't even want to think about lol

Reply:
I think that is fine. Honestly I have a hard time understanding the use case for latest. Does this make sense to you @tobymao?